packages/kbn-search-response-warnings/src/extract_warnings.ts

```diff
@@ -8,26 +8,29 @@
  */
 
 import { estypes } from '@elastic/elasticsearch';
+import type { ESQLSearchResponse } from '@kbn/es-types';
 import type { Start as InspectorStartContract, RequestAdapter } from '@kbn/inspector-plugin/public';
 import type { SearchResponseWarning } from './types';
 
 /**
  * @internal
  */
 export function extractWarnings(
-  rawResponse: estypes.SearchResponse,
+  rawResponse: estypes.SearchResponse | ESQLSearchResponse,
   inspectorService: InspectorStartContract,
   requestAdapter: RequestAdapter,
   requestName: string,
   requestId?: string
 ): SearchResponseWarning[] {
   const warnings: SearchResponseWarning[] = [];
 
+  // ES|QL supports _clusters in case of CCS but doesnt support _shards and timed_out (yet)
   const isPartial = rawResponse._clusters
     ? rawResponse._clusters.partial > 0 ||
       rawResponse._clusters.skipped > 0 ||
       rawResponse._clusters.running > 0
-    : rawResponse.timed_out || rawResponse._shards.failed > 0;
+    : ('timed_out' in rawResponse && rawResponse.timed_out) ||
```

Comment:
Is it Typescript that does not allow to simply have `rawResponse.timed_out` here?